packages/flutter/lib/src/widgets/async.dart

```diff
@@ -274,8 +291,16 @@ class AsyncSnapshot<T> {
   /// failure.
   bool get hasError => error != null;
 
+  /// Returns whether this snapshot contains a non-null [stackTrace] value.
+  ///
+  /// This is always true if the asynchronous computation's last result was
+  /// failure.
+  ///
+  /// Consequently, [hasStackTrace] is always true if [hasError] is true.
+  bool get hasStackTrace => stackTrace != null;
```

Comment:
if this is always true when hasError is true, why bother adding it?

Reply:
@Hixie Thanks! 

I thought about it as well; I should have really removed it when I switched to `StackTrace.empty`.  
I allowed `null` stack traces before, however, I realized that this would never really occur because `StreamController` and `Completer` add default stack traces and the base implementation uses `StackTrace.empty` if `null` is passed → `null` stack traces are not reasonable. So now it really is redundant. 

I will remove it and update the documentation accordingly.